core/src/main/java/org/mule/runtime/core/internal/streaming/object/factory/AbstractCursorIteratorProviderFactory.java

```diff
@@ -42,22 +43,41 @@ public AbstractCursorIteratorProviderFactory(StreamingManager streamingManager)
    * {@inheritDoc}
    */
   @Override
-  public final Object of(EventContext eventContext, Iterator iterator) {
+  public final Object of(EventContext eventContext, Iterator iterator, ComponentLocation originatingLocation) {
     if (iterator instanceof CursorIterator) {
       return streamingManager.manage(((CursorIterator) iterator).getProvider(), eventContext);
     }
 
-    Object value = resolve(iterator, eventContext);
+    Object value = resolve(iterator, eventContext, originatingLocation);
     if (value instanceof CursorProvider) {
       value = streamingManager.manage((CursorProvider) value, eventContext);
     }
 
     return value;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public final Object of(EventContext eventContext, Iterator iterator) {
+    return of(eventContext, iterator, eventContext.getOriginatingLocation());
```

Comment:
can you make the implementation null safe and pass no value instead of a wrong value for this case?